Northstar.Client/mod/scripts/vscripts/ui/menu_ns_moddownload.nut

```diff
@@ -0,0 +1,117 @@
+global function DownloadMod
+global function DisplayModDownloadErrorDialog
+
+global enum eModInstallStatus
+{
+    DOWNLOADING,
+    CHECKSUMING,
+    EXTRACTING,
+    DONE,
+    FAILED,
+    FAILED_READING_ARCHIVE,
+    FAILED_WRITING_TO_DISK,
+    MOD_FETCHING_FAILED,
+    MOD_CORRUPTED,
+    NO_DISK_SPACE_AVAILABLE,
+    NOT_FOUND
+}
+
+bool function DownloadMod( RequiredModInfo mod )
+{
+	// Downloading mod UI
+	DialogData dialogData
+	dialogData.header = Localize( "#DOWNLOADING_MOD_TITLE" )
+	dialogData.message = Localize( "#DOWNLOADING_MOD_TEXT", mod.name, mod.version )
+	dialogData.showSpinner = true;
+
+	// Prevent user from closing dialog
+	dialogData.forceChoice = true;
+	OpenDialog( dialogData )
+
+	// Save reference to UI elements, to update their content
+	var menu = GetMenu( "Dialog" )
+	var header = Hud_GetChild( menu, "DialogHeader" )
+	var body = GetSingleElementByClassname( menu, "DialogMessageClass" )
+
+	// Start actual mod downloading
+	NSDownloadMod( mod.name, mod.version )
+
+	ModInstallState state = NSGetModInstallState()
+	while (state.status < eModInstallStatus.DONE)
+	{
+		state = NSGetModInstallState()
+		UpdateModDownloadDialog( mod, state, menu, header, body )
+		WaitFrame()
+	}
+
+	print("Mod status: " + state.status)
+
+	// Close loading dialog
+	CloseActiveMenu()
+
+	return state.status == eModInstallStatus.DONE
+}
+
+const int MB = 1024*1000;
```

Comment:
Personally i would move this to the top of the file

Reply:
Addressed in 04ea7aad7b04cacd98efa61eecf698293437d4a8